### DISCUSSION TITLE
Can't run flextable_to_rmd anymore

### DISCUSSION THREAD

**Discussion by Gagaca** [davidgohel/flextable#646] [Q&A]
Hi, 
I used the flextable_to_rmd function without problem until recently. This allowed me to display flextables generated within a function, in an Rmarkdown in Word output. After several updates, I can no longer run this function which gives me the following error: 
Error in setwd(opts_knit$get("output.dir")): 
  expected character argument.

Here is an example script that does not work because of this error : 

---
output:
  word_document:
  
---
```{r setup, include=FALSE, warning=TRUE, message=TRUE}

librarian::shelf(
  'knitr', # Version 1.48 in R 4.3.2
  'flextable', # Version 0.9.6 
  'officer' # Version 0.9.6
  )
dir <- getwd()
knitr::opts_knit$set(root.dir = dir)

```
Function : 

```{r function}
fun_flextable <- function() {
  std_border <- fp_border(color = "blue")
  
  ft <- flextable(head(iris)) |> 
    border_remove() |> 
    hline(part = "all", border = std_border)
  flextable::flextable_to_rmd(ft)
}
```
Printing : 
```{r print, include=TRUE, echo=FALSE, results='asis'}
fun_flextable ()
```

How can I fix this error. Did I forget a setting?

Thank you for this excellent package, very useful for clean and flexible renderings.

**Comment by davidgohel** [comment]
Can you try https://ardata-fr.github.io/flextable-book/rendering.html#looping-in-r-mardown-documents? 

See also the original documentation https://bookdown.org/yihui/rmarkdown-cookbook/child-document.html

Your function does only print a flextable. You don't really need `flextable_to_rmd`. Why not just returning the flextable and let the usual print method do the job?
